### DISCUSSION TITLE
useFetcher / useTransition file upload progress

### DISCUSSION THREAD

**Discussion by redbaron76** [remix-run/remix#3041] [Proposals]
I've just implemented a multipart form data file upload using useFetcher `submit()` and it works very well.
Problem is that I don't have control over the progress of the upload and `useTransition` hook only gives to me if the file is going up or not. Would be possible to add to `useTransition` the **progress** of the operation in case of a `multipart/form-data` form submission? Or, in general, how is possible at the moment to get the progress of an upload operation? Thanks.

**Comment by jgb-solutions** [comment]
Was just looking for this

**Reply by mroozspace** [reply]
any updates on that?

**Reply by Bistaff** [reply]
Any updates ?

**Reply by sergiodxa** [reply]
Remix uses `fetch` to do the request and Fetch API has no way to know the progress of the request.

You could show a indefinite loading UI (like a spinner) or you would have to use XMLHttpRequest yourself for that.

**Reply by topaxi** [reply]
`fetch` can use streams as bodies, this then could be done if fetchers would allow a transform of the `body` provided to `fetch` using a custom `TransformStream` counting bytes.

**Reply by kiliman** [reply]
Here's a repo that does just that.

https://github.com/anthumchris/fetch-progress-indicators

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
